go/vumitools/middleware.py

```diff
@@ -47,16 +47,20 @@ def setup_middleware(self):
         self.strip_plus = self.config.strip_plus
 
     def handle_inbound(self, message, endpoint):
-        from_addr = normalize_msisdn(message.get('from_addr'),
-                                     country_code=self.country_code)
+        from_addr = message.get('from_addr')
+        if from_addr is not None:
+            from_addr = normalize_msisdn(from_addr,
+                                         country_code=self.country_code)
         message['from_addr'] = from_addr
```

Comment:
Should we perhaps factor this out into a `self.normalize_address(addr, strip_plus)` method that does all the necessary check and modifications and returns the new (or unmodified) address? `handle_inbound()` would pass `strip_plus=False` and `handle_outbound()` would pass `strip_plus=self.strip_plus`.